src/DonationForms/DataTransferObjects/DonationFormGoalData.php

```diff
@@ -52,23 +71,30 @@ public function __construct(int $formId, FormSettings $formSettings)
      */
     public function getCurrentAmount()
     {
-        /** @var DonationFormRepository $donationFormRepository */
-        $donationFormRepository = give(DonationFormRepository::class);
+        $query = $this->goalType->isOneOf(GoalType::SUBSCRIPTIONS(), GoalType::AMOUNT_FROM_SUBSCRIPTIONS(), GoalType::DONORS_FROM_SUBSCRIPTIONS())
+            ? new SubscriptionQuery()
+            : new DonationQuery();
+
+        $query->form($this->formId);
+
+        if($this->goalProgressType->isCustom()) {
+            $query->between($this->goalStartDate, $this->goalEndDate);
```

Comment:
@alaca @kjohnson the `$this->goalStartDate` and `$this->goalEndDate` properties are nullable from the UI.  I'm not sure MySQL will understand the null values in this between statement - so how should we handle the null values?

The expected UI behavior would translated into: 
- No start date = date created
- No end date = infinity

With that in mind, one option is to have fallback values for those scenarios.  Open for discussion 😄

Reply:
@jonwaldstein we are not passing nullable values to the DB, we were converting the dates to timestamps and we used `date` to format the timestamp to MySQL date format. If the dates are empty or invalid, `strtotime` will fallback to the 1st of January 1970. This works for the start date, but for the end date, we should set the current date so that we have a _correct_ date range.

Anyway, here is the commit where I fixed it https://github.com/impress-org/givewp/pull/7384/commits/ad2bbb364335b362ef7dff7662d9f9bd216334ef

Reply:
@alaca thanks!